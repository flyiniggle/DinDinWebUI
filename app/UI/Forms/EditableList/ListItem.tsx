import * as React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faMinus } from '@fortawesome/free-solid-svg-icons';


interface IListItem {
    text: string
    key: number,
    handleRemove: (number) => void
}

function ListItem(props: IListItem) {
    return (
        <div className="list-item">
            {props.text}
            <span onClick={() => {
                props.handleRemove(props.key)
            }}>
                <FontAwesomeIcon icon={faMinus} />
            </span>
        </div>
    )
}

export default ListItem
export { IListItem };