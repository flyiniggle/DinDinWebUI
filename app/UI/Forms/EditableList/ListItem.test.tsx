import * as React from 'react';
import { shallow } from 'enzyme';
import ListItem, { IListAdder } from 'UI/Forms/EditableList/ListItem';

describe('#UI #Forms #EditableList #ListItem', function() {
    it('should render', function () {
        const props: IListAdder = {
            text: 'test',
            id: 1,
            handleRemove: () => null
        }
        const wrapper = shallow(<ListItem {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should display the provided text.', function () {
        const props: IListAdder = {
            text: 'test',
            id: 1,
            handleRemove: () => null
        }
        const wrapper = shallow(<ListItem {...props} />);
        const textContainer = wrapper.find('.form-control');

        expect(textContainer.text()).toEqual(props.text);
    })

    it('should call handleRemove with the id prop when the remove button is clicked.', function () {
        const spy = jest.fn()
        const props: IListAdder = {
            text: 'test',
            id: 1,
            handleRemove: spy
        }
        const wrapper = shallow(<ListItem {...props} />);

        wrapper.find('.input-group-append').simulate('click', new Event('click'))

        expect(spy).toHaveBeenCalledWith(props.id);
    })
});