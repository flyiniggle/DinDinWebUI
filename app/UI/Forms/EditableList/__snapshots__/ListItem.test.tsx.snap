// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#UI #Forms #EditableList #ListItem should render 1`] = `
<div
  className="list-item input-group mb-2"
>
  <span
    className="form-control form-control-sm"
  >
    test
  </span>
  <span
    className="input-group-append"
    onClick={[Function]}
  >
    <AsyncButton
      className="btn btn-sm btn-outline-primary"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              448,
              512,
              Array [],
              "f068",
              "M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z",
            ],
            "iconName": "minus",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        transform={null}
      />
    </AsyncButton>
  </span>
</div>
`;
